### PR TITLE
docs: metrics

### PR DESCRIPTION
**Pull request by technicallyty**
adds docs for:

- how to setup metrics
- reference for all metrics (app and connect)

### CODE REVIEW

File: docs/metrics/application-reference.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+description: Reference for the available metrics in the Application
+title: Application
+icon: link
+---
+
+## Exposed Metrics
+By default, Cosmos SDK application expose metrics on port `26660`. These metrics are in the Prometheus format and can be scraped by any monitoring system that supports Prometheus format.
```
Comment (aljo242) [inline docs/metrics/application-reference.mdx]:
```suggestion
## Exposed Metrics

By default, Cosmos SDK applications expose metrics on port `26660`. These metrics are in the Prometheus format and can be scraped by any monitoring system that supports Prometheus format.
```

Comment (aljo242) [reply docs/metrics/application-reference.mdx]: would link to prometheus docs here

File: docs/metrics/application-reference.mdx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Oracle Client Connection Metrics
+
+- **oracle_response_latency:** Histogram that measures the time in nanoseconds between request/response of from the Application to Oracle.
+- **oracle_responses:** Counter that measures the number of oracle responses.
+
+## ABCI Metrics
```
Comment (aljo242) [inline docs/metrics/application-reference.mdx:16]: nit: header paragraph spacing

File: docs/metrics/oracle-reference.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+description: Reference for the available metrics in the Connect oracle
+title: Oracle
+icon: book-sparkles
+---
+
+## Dashboard Terminology
+
+- **Market:** Pair of assets that are traded against each other. For example, the BTC-USD market is the market where Bitcoin is traded against the US Dollar.
```
Comment (aljo242) [inline docs/metrics/oracle-reference.mdx]:
```suggestion
- **Market:** Pair of assets that are traded against each other. For example, the BTC/USD market is the market where Bitcoin is traded against the US Dollar.
```

we always refer to markets with a "/" delimiter

File: docs/metrics/oracle-reference.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- **side_car_health_check_ticker_updates_total:** Counter that increments every time the side-car updates the price of a given market. This is a good indicator of the overall health of a given market.
+- **side_car_health_check_provider_updates_total:** Counter that increments every time the side-car utilizes a given providers market data. This is a good indicator of the health of a given provider. Note that providers may not be responsible for every market. However, the sidecar correctly tracks the number of expected updates for each provider.
+
+
+### Price Metrics
+
+Connect exposes various metrics related to market prices. These metrics are useful for monitoring the health of the price feeds and the aggregation process. The rate of updates for these metrics is likely going to be much greater than the health metrics. This is expected as we poll the providers much more frequently than we update all prices! As such, the metrics below are not expected to be updated at the same rate as the health metrics. However, they are greatly useful to quickly identify issues with the price feeds.
```
Comment (aljo242) [inline docs/metrics/oracle-reference.mdx]: unsure about using exclamation points

File: docs/metrics/oracle-reference.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+### HTTP Metrics
+
+Connect exposes various metrics related to HTTP requests made by the sidecar - including the number of requests, the response time, and the status code. These metrics can be used to monitor the health of the sidecar's HTTP endpoints.
+
+- **side_car_api_http_status_code:** The status codes of the HTTP response made by the side-car.
+- **side_car_api_response_latency_bucket:** The response latency of the HTTP requests made by the side-car.
+
+### WebSocket Metrics
+
+Connect exposes various metrics related to websocket connections made by the sidecar. These metrics can be used to monitor the health of the sidecar's WebSocket connections.
```
Comment (aljo242) [inline docs/metrics/oracle-reference.mdx]: We should consistently format "WebSocket"

should always be capitalized like that ^^^

File: docs/metrics/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+---
+description: Overview of metrics in Connect
+title: Overview
+icon: ruler
+---
+
+The Connect sidecar is equipped to emit prometheus metrics to provide observability into the health of the oracle sidecar.
+
+### Metrics Emitted by Connect
```
Comment (aljo242) [inline docs/metrics/overview.mdx:9]: spacing

File: docs/metrics/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+---
+description: Overview of metrics in Connect
+title: Overview
+icon: ruler
+---
+
+The Connect sidecar is equipped to emit prometheus metrics to provide observability into the health of the oracle sidecar.
```
Comment (aljo242) [inline docs/metrics/overview.mdx]:
```suggestion
The Connect sidecar is equipped to emit Prometheus metrics to provide observability into the health of the oracle sidecar.
```

should consistently capitalize "Prometheus"

File: docs/metrics/overview.mdx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+- Number of Oracle responses
+- ABCI method latency
+- Size of messages
+- Aggregate ticker prices written to app state
+- Prices reported by each validator
+- Status of reports from each validator
+
+
+### Need More Metrics?
+If you find yourself wanting more observability than what is provided, please open an issue on our [GitHub repo](https://github.com/skip-mev/slinky)
```
Comment (aljo242) [inline docs/metrics/overview.mdx]:
```suggestion
If you find yourself wanting more observability than what is provided, please open an issue on our [GitHub repo](https://github.com/skip-mev/slinky).
```

File: docs/metrics/setup.mdx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+This document will guide you on how to utilize Connect's metrics.
+
+## Prerequisites
+- You have a [Prometheus](https://prometheus.io/) instance
+- You have a [Grafana](https://grafana.com/) instance
+- You've configured a data source in Grafana for your Prometheus instance
+- You have set the metrics [configuration](/validators/configuration#env-vars) in the Connect sidecar
+
+If you're not sure how to set up Prometheus and Grafana, see this [document](https://grafana.com/docs/grafana/latest/getting-started/get-started-grafana-prometheus/).
```
Comment (aljo242) [inline docs/metrics/setup.mdx]:
```suggestion
If you're not sure how to set up Prometheus and Grafana, see this [guide](https://grafana.com/docs/grafana/latest/getting-started/get-started-grafana-prometheus/).
```

File: docs/metrics/setup.mdx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+## Import Dashboards
+
+Import the following dashboard JSON files to view the available metrics in Grafana. If you do not know how to import a Grafana dashboard, see this [document](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/import-dashboards/).
```
Comment (aljo242) [inline docs/metrics/setup.mdx]:
```suggestion
Import the following dashboard JSON files to view the available metrics in Grafana. If you do not know how to import a Grafana dashboard, see this [guide](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/import-dashboards/).
```

File: docs/metrics/oracle-reference.mdx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+Connect also exposes metrics on the `/metrics` endpoint on port `8002` by default. These metrics are in the Prometheus format and can be scraped by any monitoring system that supports Prometheus format.
+
+### Health Metrics
+
+The following health metrics are emitted in Connect.
+
+- **side_car_health_check_system_updates_total:** Counter that increments every time the sidecar updates its internal state. This is a good indicator of the sidecar's overall health.
+- **side_car_health_check_ticker_updates_total:** Counter that increments every time the side-car updates the price of a given market. This is a good indicator of the overall health of a given market.
+- **side_car_health_check_provider_updates_total:** Counter that increments every time the side-car utilizes a given providers market data. This is a good indicator of the health of a given provider. Note that providers may not be responsible for every market. However, the sidecar correctly tracks the number of expected updates for each provider.
```
Comment (davidterpay) [inline docs/metrics/oracle-reference.mdx]: I would say something about this being recommended but not required, `side_car_health_check_provider_updates_total` can be a bit noisy.

**Review by davidterpay — APPROVED** [review]

File: docs/metrics/oracle-reference.mdx (hunk 11 above)
Comment (technicallyty) [reply docs/metrics/oracle-reference.mdx]: [notify about noisyness](https://github.com/skip-mev/slinky/pull/616/commits/25ebef9f0430bfa947cbf2e240fc5a4bdfdc57b5) added a snippet about it being noisy and to consider omitting it

File: docs/metrics/overview.mdx (hunk 6 above)
Comment (technicallyty) [reply docs/metrics/overview.mdx:9]: fixed [spacing](https://github.com/skip-mev/slinky/pull/616/commits/25c442a529d20c190d113fc9e2c569210e7888c2)

File: docs/metrics/application-reference.mdx (hunk 2 above)
Comment (technicallyty) [reply docs/metrics/application-reference.mdx:16]: fixed [spacing](https://github.com/skip-mev/slinky/pull/616/commits/25c442a529d20c190d113fc9e2c569210e7888c2)

File: docs/metrics/oracle-reference.mdx (hunk 4 above)
Comment (technicallyty) [reply docs/metrics/oracle-reference.mdx]: removed [52508946d189ac984ca3b49ff00c77d4ae88846f](https://github.com/skip-mev/slinky/pull/616/commits/52508946d189ac984ca3b49ff00c77d4ae88846f)

File: docs/metrics/oracle-reference.mdx (hunk 5 above)
Comment (technicallyty) [reply docs/metrics/oracle-reference.mdx]: fixed [cap websockets](https://github.com/skip-mev/slinky/pull/616/commits/4a5000116f7fe1a1a70aee66ff140b0c91f6e4cb)

File: docs/metrics/application-reference.mdx (hunk 1 above)
Comment (technicallyty) [reply docs/metrics/application-reference.mdx]: done [link to prom docs](https://github.com/skip-mev/slinky/pull/616/commits/ebb188b77e4ef2bbbe88b7c1892b7322ba2523cc)

File: docs/metrics/oracle-reference.mdx [hunk 12]
```diff
@@ -0,0 +1,51 @@
+---
+description: Reference for the available metrics in the Connect oracle
+title: Oracle
+icon: display-chart-up
+---
+
+## Dashboard Terminology
+
+- **Market:** Pair of assets that are traded against each other. For example, the BTC/USD market is the market where Bitcoin is traded against the US Dollar.
+- **Price Feed:** A price feed is indexed by a price provider and a market. For example, the Coinbase API provides a price feed for the BTC-USD market.
```
Comment (aljo242) [inline docs/metrics/oracle-reference.mdx]:
```suggestion
- **Price Feed:** A price feed is indexed by a price provider and a market. For example, the Coinbase API provides a price feed for the BTC/USD market.
```

let's make sure these are all consistently formatted

**Review by aljo242 — APPROVED** [review]

**Pull request merged**